app/pathogen/arbovirus/analyze/recharts/change-in-median-seroprevalence-over-time-graph.tsx

```diff
@@ -0,0 +1,196 @@
+  let currentIntervalStartDate = startOfYear(input.earliestDate);
+  const buckets: TimeInterval[] = [];
+  
+  while(isBefore(currentIntervalStartDate, input.latestDate)) {
+    const currentIntervalEndDate = endOfYear(addYears(currentIntervalStartDate, input.groupingBucketSize.years));
+    const bucketToAdd = {
+      intervalStartDate: parseISO(currentIntervalStartDate.toISOString()),
+      intervalEndDate: parseISO(currentIntervalEndDate.toISOString())
+    }
+
+    buckets.push(bucketToAdd);
+
+    currentIntervalStartDate = startOfYear(addYears(currentIntervalEndDate, 1));
+  }
+
+  return {
+    buckets
+  }
+}
+
+const groupDataPointsIntoTimeBuckets = <
+  TDataPoint extends GroupableIntoTimeBuckets
+>(
+  input: GroupDataPointsIntoTimeBucketsInput<TDataPoint>
+): GroupDataPointsIntoTimeBucketsOutput<TDataPoint> => {
+  const earliestDate = input.dataPoints
+    .toSorted((a, b) => isBefore(a.groupingTimeInterval.intervalStartDate, b.groupingTimeInterval.intervalStartDate) ? -1 : 1)
+    [0]?.groupingTimeInterval.intervalStartDate;
+
+  const latestDate = input.dataPoints
+    .toSorted((a, b) => isAfter(a.groupingTimeInterval.intervalEndDate, b.groupingTimeInterval.intervalEndDate) ? -1 : 1)
+    [0]?.groupingTimeInterval.intervalEndDate;
+
+  if (!earliestDate || !latestDate) {
+    return {
+      groupedDataPoints: [],
+    };
+  }
+
+  const { buckets } = generateBucketsForTimeInterval({
+    earliestDate,
+    latestDate,
+    groupingBucketSize: input.groupingBucketSize
+  })
+
+  let groupedDataPoints = buckets.map((bucket) => ({
+    interval: bucket,
+    dataPoints: input.dataPoints.filter((dataPoint) => doTimeIntervalsOverlap(bucket, dataPoint.groupingTimeInterval))
+  })).filter((bucketAndDataPoints) => bucketAndDataPoints.dataPoints.length > 0)
+
+  return {
+    groupedDataPoints,
+  };
+};
+
+export const ChangeInMedianSeroprevalenceOverTimeGraph = (): React.ReactNode => {
+  const state = useContext(ArboContext);
+
+  const dataGroupedByArbovirus = typedGroupBy(
+    state.filteredData,
+    (dataPoint) => dataPoint.pathogen
+  );
+
+  const dataGroupedByArbovirusThenByTimeBucket = typedObjectFromEntries(
+    typedObjectEntries(dataGroupedByArbovirus).map(
+      ([arbovirus, dataPointsForAParticularArbovirus]) => [
+        arbovirus,
+        groupDataPointsIntoTimeBuckets({
+          dataPoints: dataPointsForAParticularArbovirus.map((dataPoint) => ({
+            ...dataPoint,
+            groupingTimeInterval: {
+              intervalStartDate: parseISO(dataPoint.sampleStartDate),
+              intervalEndDate: parseISO(dataPoint.sampleEndDate),
+            },
+          })),
+          groupingBucketSize: {
+            years: 5,
+          },
+        }).groupedDataPoints,
+      ]
+    )
+  );
+
+  return (
+    <div className="h-full flex flex-col">
+      <div className="h-full flex flex-row flex-wrap">
+        {Object.keys(dataGroupedByArbovirusThenByTimeBucket).map((arbovirus, index) => {
```

Comment:
Fair amount of copy-pistachio from "Median seroprevalence by WHO region and age group" if you recognize some of this.